File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/SamplesSource.java
/**
 *
 * @file
 *
 * @brief Samples source interface
 *
 * @author vitamin.caig@gmail.com
 *
 */

package app.zxtune.sound;

import app.zxtune.TimeStamp;

public interface SamplesSource {

  final class Channels {
    public static final int COUNT = 2;
  }

  final class Sample {
    public static final int BYTES = 2;
  }
  
  /**
   * Acquire next sound chunk
   * @param buf result buffer of 16-bit signed interleaved stereo signal
   * @return true if buffer filled, else reset position to initial
   */
  boolean getSamples(short[] buf);

  /**
   * Synchronously changes playback position
   * @param pos
   * @throws Exception
   */
  void setPosition(TimeStamp pos);

  /**
   * Get current playback position
   * @return pos
   * @throws Exception
   */
  TimeStamp getPosition();
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/SamplesSource.kt
/**
 *
 * @file
 *
 * @brief Samples source interface
 *
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.sound

import app.zxtune.TimeStamp

interface SamplesSource {
    object Channels {
        const val COUNT = 2
    }

    object Sample {
        const val BYTES = 2
    }

    /**
     * Acquire next sound chunk
     * @param buf result buffer of 16-bit signed interleaved stereo signal
     * @return true if buffer filled, else reset position to initial
     */
    fun getSamples(buf: ShortArray): Boolean

    /**
     * Current playback position
     */
    var position: TimeStamp
}
